components/raftstore/src/store/worker/life.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright 2024 TiKV Project Authors. Licensed under Apache-2.0.
+
+use engine_traits::{Engines, KvEngine, RaftEngine};
+
+/// An assistant to help manage the lifecycle of engines.
+///
+/// Currently, it only enables manual compaction jobs before starting the
+/// engines and disables manual compaction jobs before shutting down the
+/// engines.
+pub struct Assistant<E, R>
```

Comment:
The code under worker typically the long running tasks that run under background threads. 
But in this case, we need a function to stop manual compaction. I think it's a bit overkill to have a specialized worker for it.  I think to have an Assistant is fine, but it's not the worker IMO.